tests/unittests/ImageTest.cpp

```diff
@@ -55,6 +55,39 @@ TEST(Image, readBadImages) {
   ASSERT_FALSE(loadSuccess);
 }
 
+TEST(Image, readPngImageAndPreprocessWithAndWithoutInputTensor) {
+  auto image1 = readPngImageAndPreprocess(
+      "tests/images/imagenet/cat_285.png", ImageNormalizationMode::k0to1,
+      ImageChannelOrder::RGB, ImageLayout::NHWC, imagenetNormMean,
+      imagenetNormStd);
+  Tensor image2;
+  readPngImageAndPreprocess(image2, "tests/images/imagenet/cat_285.png",
+                            ImageNormalizationMode::k0to1,
+                            ImageChannelOrder::BGR, ImageLayout::NCHW,
+                            imagenetNormMean, imagenetNormStd);
+
+  // Test if the preprocess actually happened.
+  size_t imgHeight = image1.dims()[0];
+  size_t imgWidth = image1.dims()[1];
+  size_t numChannels = image1.dims()[2];
+
+  Tensor swizzled(image1.getType());
```

Comment:
This code code seems to be identical to the preprocessing code in `glow::readPngImageAndPreprocessHelper`. I'm not sure what the value of this is because a bug in both places won't get caught and a bug that is introduced into `readPngImageAndPreprocess` is likely to be hidden by copying that code here whenever it is updated.

Reply:
sounds right. I modified the test case that I move the transpose operation on image2 instead to make image2 become NHWC and do the RBG change to image1, then i can check whether image1 equals image2. 

It should be same logic but the operations are not same as the `readPngImageAndPreprocess()` implementation